app/bundles/LeadBundle/Controller/LeadController.php

```diff
@@ -1394,10 +1394,48 @@ public function emailAction($objectId = 0)
                             empty($email['fromname']) ? null : $email['fromname']
                         );
 
+                        // Attachments
+                        $assets = [];
+                        $assetsIds  = (!empty($email['assetAttachments'])) ? $email['assetAttachments'] : [];
+                        foreach($assetsIds as $assetId) {
+                            $model = $this->factory->getModel('asset');
```

Comment:
`$this->factory` is deprecated-. You should use `$this->getModel('asset.asset')` or  `$this->get('mautic.asset.model.asset')`. Or better use initialize method allow use variable enywhere same like here https://github.com/mautic/mautic/blob/b05c93c2d96ca04220357fc8b2bee53e67469701/app/bundles/FormBundle/Controller/FieldController.php#L42 

Also we don't need call model in foreach cycle few times. We can move it above or to initialize method